### PR TITLE
[LWDM] feat(common): [LIVE-29431] add ensure-app-ready use case for DIE initialization

### PR DESCRIPTION
**Pull request by OlivierFreyssinet**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.**
- [x] **Impact of the changes:**
  - Adds a new DMK-native "ensure app ready" use case in `live-common` and a new `EnsureAppReadyDeviceAction` in `live-dmk-shared`. Nothing replaces the legacy `hw/actions/app` flow yet — the new code path is wired into the Device Intent Executor only, and exercised through the LWM **Debug → Device Intent Executor → Initialization** screen. QA should focus on this debug screen.
  - **Modifies the existing `ConnectAppDeviceAction`** (`libs/live-dmk-shared/src/device-action/ConnectApp/ConnectAppDeviceAction.ts`) — see [Changes to `ConnectAppDeviceAction`](#changes-to-connectappdeviceaction) below. The action is also instantiated by the legacy DMK-gated path (`libs/ledger-live-common/src/hw/connectApp.ts`, behind the `isLdmkConnectAppEnabled` flag) and by `apps/wallet-cli`, so the change is visible there too. The legacy `ConnectAppEventMapper` doesn't consume the new `currentApp` output field, so for legacy callers the behavioural delta is essentially **one extra `getStatus` round-trip before `Success`**. QA should retest standard connect-app flows on the legacy harness (open / install / wrong app / unlock / device locked / refused-by-user) on both Bluetooth and USB to make sure nothing regresses.
  - On LWM, the only user-visible artifact is the new debug Initialization Playground that renders the raw states. No production flow UI changed.

#### Context — what the legacy "connect app" does

Almost every flow in Ledger Wallet that talks to the device (send, receive, swap, add account, …) starts with the same step: making sure the device is in a state where it can actually run the requested operation. That step is the **"connect app"** step. Today it lives in `libs/ledger-live-common/src/hw/connectApp.ts` and is driven by the `useAppDeviceAction` hook in `libs/ledger-live-common/src/hw/actions/app.ts`.

Flows themselves provide very little: just an `AppRequestInput` describing the domain context they care about — the **account**, **currency**, **token currency**, optional **dependencies** (other apps that must be installed alongside), and a couple of behavioural flags (`requireLatestFirmware`, `allowPartialDependencies`). They don't pick the app name, they don't know about min app versions, they don't know about deprecation rules.

Everything else is **derived internally** by the connect-app logic from that input + remote configuration it pulls itself:

1. it resolves the actual **app name** to open (and the dependency apps) from the currency / token / account,
2. it resolves whether a **derivation** is required for that input, and on which path / derivation mode,
3. it pulls the **minimum app version** and the **deprecation configuration** (currently used to warn users — or block them — on Nano S for currencies whose support is being phased out) from internal sources,
4. it then checks if the right app is already opened on the device, otherwise opens it (or **installs / updates** it through the manager),
5. it surfaces the appropriate UI when constraints fail (min app version, device deprecation, out of storage, unsupported feature, …),
6. when a derivation is required, it derives the address and **checks it against the expected account** so we don't sign with the wrong device (the "wrong device for account" guard).

#### Why we are rebuilding it

The current implementation in `libs/ledger-live-common/src/hw/actions/app.ts` (+ `connectApp.ts`) has several structural issues:

- **Not DMK-native**: it is built on top of `withDevice` / registered transport modules (`registerTransportModule`, used e.g. in `apps/ledger-live-mobile/src/services/registerTransports.ts`), which are being phased out in favour of the Device Management Kit (DMK) sessions.
- **One giant reducer / one giant state**: the state object (see `State` in `libs/ledger-live-common/src/hw/actions/app.ts`) merges the possible states of *all* device actions (connect app + every flow-specific device action). The host apps then render that state through huge mega-components — currently:
  - `apps/ledger-live-mobile/src/components/DeviceAction/index.tsx` (~810 lines) + `rendering.tsx` (~1.3k lines),
  - `apps/ledger-live-desktop/src/renderer/components/DeviceAction/index.tsx` (~850 lines) + `rendering.tsx` (~1.5k lines).

  Each branches on dozens of booleans (`opened`, `requestQuitApp`, `requestOpenApp`, `unresponsive`, `requiresAppInstallation`, `displayUpgradeWarning`, …). Adding or tweaking a state is risky.
- **The UI is also doing domain-level work**, for two reasons that fall out of the previous point:
  - **Low-level error discrimination at the view layer**: because the legacy state only carries a generic `error: Error | null`, the UI itself has to figure out what each error actually means by doing `instanceof` checks against transport-/HW-level error classes (`UserRefusedOnDevice`, `LatestFirmwareVersionRequired`, `UnsupportedFeatureError`, `NoSuchAppOnProvider`, `DeviceNotOnboarded`, `TransportStatusError`, …). See `apps/ledger-live-mobile/src/components/DeviceAction/index.tsx` (e.g. lines ~527 / ~642-660) and the desktop equivalent.
  - **Business logic at the view layer**: the device-deprecation flow (Nano S phase-out) is partly implemented inside the legacy `DeviceAction/index.tsx` — it reads `settingsState.deprecationDoNotRemind`, decides which deprecation screens to show (warning / clear-signing / blocking error), in which order, and calls back into `deviceDeprecationRules.onContinue(...)` accordingly (`apps/ledger-live-mobile/src/components/DeviceAction/index.tsx` lines ~339-399). That is policy that has nothing to do with rendering.
- **Constraints are not injectable**: `connectApp.ts` reads `getMinVersion` and `getDeprecationConfig` directly from `libs/ledger-live-common/src/apps/support.ts` (which themselves read from the remote config). There is no seam to override them, so it is essentially impossible to test edge cases without mutating a remote config.

The plan is to introduce the **Device Intent Executor (DIE)**, which isolates:

- **Connection** + **device-context initialization** ("ensure app ready") on one side, with their own dedicated UI,
- Flow-specific logic ("device intents") on the other side, with their own dedicated business logic and UI per intent.

**This PR delivers the second half of the initialization phase: the new "ensure app ready" use case, with full feature parity with the legacy connect-app flow, but DMK-native, configuration-injectable, and built around a single state stream the UI can render directly.**

##### High-level architecture

The new implementation is structured as one public use case (`ensureAppReadyUseCase`) that exposes a stream of UI-ready states. Everything the UI needs to render lives in those states — there is no domain logic in the view.

```mermaid
flowchart LR
    UI["LWM<br/>DeviceContextInitializerComponentLWM<br/>(View + ViewModel)"]
    UC["ensureAppReadyUseCase<br/>(live-common)"]
    EAR["EnsureAppReadyDeviceAction<br/>(live-dmk-shared)"]
    CA["ConnectAppDeviceAction<br/>(live-dmk-shared)"]
    DMK[(DMK / Device)]

    UI -- "subscribe" --> UC
    UC -- "Observable&lt;EnsureAppReadyState&gt;" --> UI
    UC -- "builds + injects config" --> EAR
    EAR -- "wraps" --> CA
    CA -- "executes" --> DMK
```

Where to look:

- LWM view + view-model: `apps/ledger-live-mobile/src/mvvm/components/DeviceIntentExecutor/DeviceContextInitializerComponentLWM/`
- Use case: `libs/ledger-live-common/src/device/use-cases/ensureAppReady/ensureAppReadyUseCase.ts`
- Device action wrapper: `libs/live-dmk-shared/src/device-action/EnsureAppReady/EnsureAppReadyDeviceAction.ts`
- Underlying connect-app DA (extended in this PR): `libs/live-dmk-shared/src/device-action/ConnectApp/ConnectAppDeviceAction.ts`

The view-side contract is intentionally trivial:

```mermaid
sequenceDiagram
    participant V as View (LWM)
    participant VM as ViewModel
    participant UC as ensureAppReadyUseCase
    V->>VM: mount
    VM->>UC: subscribe(input, sideEffects, deps)
    UC-->>VM: state #1 (Loading)
    VM-->>V: render state
    UC-->>VM: state #2 (DeviceLocked, retry)
    VM-->>V: render state
    UC-->>VM: state #N (Success, extractedContext)
    VM->>V: onContextInitialized(extractedContext)
```

The ViewModel (`useDeviceContextInitializerComponentLWMViewModel.ts`) runs the use case in a `useEffect`, stores each emitted state in `useState`, and forwards the final extracted context once a `Success` state is observed. The View (`DeviceContextInitializerComponentLWMView.tsx`) is a pure switch over `EnsureAppReadyState.type`.

##### State design

The full list of states (and the categorization enums) is defined here:

- [`libs/live-dmk-shared/src/device-action/EnsureAppReady/state.ts`](https://github.com/LedgerHQ/ledger-live/blob/feat/live-29431-connect-app-init-job/libs/live-dmk-shared/src/device-action/EnsureAppReady/state.ts)

The snapshot → state translation logic lives in `libs/live-dmk-shared/src/device-action/EnsureAppReady/stateMapping.ts` (pending → user-interaction / installing-app states; errors → blocking / retryable / unexpected-error states).

This is what addresses the "UI doing domain-level work" point from above: every meaningful situation (including the ones the legacy flow exposed as raw errors and the ones implemented as ad-hoc UI logic) is a **dedicated state with all the data the screen needs**. The view layer doesn't `instanceof` errors and doesn't implement the deprecation-flow policy anymore — it just renders.

States are grouped into clear categories:

| Category | States | Meaning |
| --- | --- | --- |
| **Loading** | `Loading`, `InstallingApp` | Work in progress, no user input expected. |
| **Device interaction required** | `UnlockDevice`, `AllowSecureConnection`, `ConfirmOpenApp` | The device is asking the user to do something. |
| **App interaction required** | `DeviceDeprecatedNonBlocking`, `OutdatedAppWarning` | The user must acknowledge a warning (`onContinue`) before proceeding. |
| **Retryable** | `DeviceLocked`, `UserRefusedOnDevice`, `DeviceBusy` | A re-attempt can fix it; state exposes a `retry()` callback. |
| **Blocking (final, non-retryable)** | `WrongDeviceForAccount`, `DeviceDeprecatedBlocking`, `UnsupportedFirmwareVersion`, `UnsupportedApplication`, `UnsupportedFeature`, `DeviceOutOfStorageSpace`, `DeviceNotOnboarded` | A user-actionable but non-recoverable situation; carries the data the screen needs. |
| **Final** | `Success` (with `extractedContext`) / `Error` (with the unexpected error) | Stream completes after this. |

Notably:

- "Wrong device for account" and the deprecation cases are first-class states (with the data the screen needs to render), not raw errors to translate at the UI level.
- Retryable states embed their own `retry()` callback, so the UI doesn't need to know how to restart anything.
- Warning states embed their own `onContinue()` callback to resume the initialization flow if the user decides to, mirroring the same approach.

##### Internal flow

```mermaid
flowchart TB
    Input["AppRequestInput<br/>{ account, currency, token,<br/>dependencies, requireLatestFirmware }"]
    Build["buildEnsureAppReadyInput<br/>(parity with legacy)"]
    EARInput["EnsureAppReadyInput<br/>{ appName, dependencies,<br/>requiresDerivation, expectedAccount,<br/>deprecation }"]
    UC["ensureAppReadyUseCase"]
    Build2["buildConnectAppDeviceActionInput<br/>(injects getMinVersion +<br/>getDeprecationConfig)"]
    CADA["ConnectAppDeviceAction"]
    EARDA["EnsureAppReadyDeviceAction<br/>(wraps CA, adds wrong-device<br/>+ outdated-app + final mapping)"]
    Stream["Observable&lt;EnsureAppReadyState&gt;"]

    Input --> Build --> EARInput --> UC
    UC --> Build2 --> CADA
    UC --> EARDA
    CADA -. snapshots .-> EARDA
    EARDA --> Stream
```

Concretely:

1. **Input translation** — `buildEnsureAppReadyInput` (`libs/ledger-live-common/src/device/use-cases/ensureAppReady/buildEnsureAppReadyInput.ts`) translates the legacy-shaped `AppRequestInput` (account / currency / token / dependencies) into the DIE-shaped `EnsureAppReadyInput`. App-name / dependency-app-name / required-derivation resolution is delegated to `resolveAppRequestRequirements` (`libs/ledger-live-common/src/hw/deviceInitialization/helpers/resolveAppRequestRequirements.ts`, also reused by the legacy flow). It additionally derives `expectedAccount` (for the wrong-device guard, via `buildExpectedAccountIdentity`) and `deprecation` (flow + currency name). Callers keep the same input shape they already use today, so flow migration is mechanical.
2. **Use-case wiring** — `ensureAppReadyUseCase` (`libs/ledger-live-common/src/device/use-cases/ensureAppReady/ensureAppReadyUseCase.ts`) wraps each attempt with `withRetryableRepeat` and delegates the actual run to `runEnsureAppReadyAttempt` (`helpers/runEnsureAppReadyAttempt.ts`). The attempt:
   - builds the `ConnectAppDAInput` via `buildConnectAppDeviceActionInput` (`helpers/buildConnectAppDAInput.ts`), which is where the injected `getMinVersion` / `getDeprecationConfig` are actually applied,
   - instantiates a `ConnectAppDeviceAction`, then hands it to a new `EnsureAppReadyDeviceAction`,
   - registers `ConnectAppSideEffectsHandler` (`helpers/ConnectAppSideEffectsHandler.ts`) as an additional snapshot handler so Live-specific side effects fire alongside snapshot mapping,
   - executes via `dmk.executeDeviceAction(...)` and forwards unexpected device-action errors as a final `Error` state.
3. **Wrapper device action** — `EnsureAppReadyDeviceAction` (`libs/live-dmk-shared/src/device-action/EnsureAppReady/EnsureAppReadyDeviceAction.ts`) is an xstate-based wrapper around `ConnectAppDeviceAction` that:
   - drives a small state machine `ConnectApp → AfterConnectApp → AfterGetAppAndVersion → (OutdatedAppWarning?) → Finalize → Terminated`,
   - re-emits each `ConnectAppDeviceAction` snapshot as an `EnsureAppReadyState` through `EnsureAppReadyStateEmitter` (`EnsureAppReadyStateEmitter.ts`), which delegates to `stateMapping.ts`,
   - captures completion data (device metadata, derivation, current app) via `ConnectAppCompletionCapturer` (`ConnectAppCompletionCapturer.ts`),
   - performs the **outdated app warning** check before completion using the injected `shouldUpgrade` dependency, exposing an `onContinue()` callback,
   - emits the **final state** via the injected `buildFinalState` dependency, which on the live-common side does the **wrong device for account** check (`libs/ledger-live-common/src/device/use-cases/ensureAppReady/helpers/buildFinalState.ts` → `validateDerivedAddress`).
4. **Side effects** — `ConnectAppSideEffectsHandler` observes each `ConnectAppDAState` and fires `onDeviceIdObserved` (from `intermediateValue.deviceId`) and `onLastSeenDeviceInfoObserved` (from `intermediateValue.deviceMetadata`). On LWM, those map to the identities slice and `setLastSeenDeviceInfo`. They were previously executed implicitly by the legacy harness; here they are explicit, testable, and isolated.
5. **Retry** — `withRetryableRepeat` (`helpers/withRetryableRepeat.ts`) wraps each attempt: when the latest emitted state is retryable and its `retry()` callback is invoked, the previous attempt is torn down and a fresh attempt starts (re-emitting `Loading` first). Calling `retry()` more than once during the same attempt is a no-op.

##### Changes to `ConnectAppDeviceAction`

`libs/live-dmk-shared/src/device-action/ConnectApp/ConnectAppDeviceAction.ts` is the existing DMK device action that already does the heavy lifting (open / install / update / required derivation / deprecation rules). Two small but meaningful changes were needed for the new wrapper to work correctly, plus a small refactor:

- **`currentApp` is refreshed before `Success`.** A new `RefreshDeviceStatus` step is inserted between the existing terminal branches (after install, after open-with-deps, after `GetDerivation`) and `Success`. It re-runs the existing `getStatus` actor so that the `currentApp` (`name`, `version`) returned in the action's `Output` reflects what is **actually opened on the device** at the very end. Without this step, the wrapper's outdated-app check (`shouldUpgrade(currentApp.name, currentApp.version)`) would race against stale data when an install/update/open just happened. The `Output` type now exposes `currentApp?: GetAppAndVersionResponse` (`libs/live-dmk-shared/src/device-action/ConnectApp/types.ts`), which the wrapper consumes via `ConnectAppCompletionCapturer`.
- **`deviceMetadata` is exposed in `intermediateValue`** (in addition to the final output). Snapshot handlers — including the new `ConnectAppSideEffectsHandler` (last-seen device info) — need to react to it as soon as it's available, because `ConnectAppDeviceAction` can still error after the metadata has been fetched, in which case the wrapper would never see the metadata if it only came through the final output. `ConnectAppDAIntermediateValue` now has `deviceMetadata: GetDeviceMetadataDAOutput | undefined`.
- **Refactor**: extracted the shared `assignDeviceStatusFromOutput` xstate action, reused by both `IsDeviceDeprecatedCheck` (existing) and the new `RefreshDeviceStatus` step.

These changes are also seen by the legacy DMK-gated path (`libs/ledger-live-common/src/hw/connectApp.ts` line 584, behind `isLdmkConnectAppEnabled`) and by `apps/wallet-cli`. The legacy `ConnectAppEventMapper` doesn't read the new `currentApp` output field (it pulls `currentApp` from the device session state), and the new `intermediateValue.deviceMetadata` is purely additive — so the only behavioural delta for legacy callers is one extra `getStatus` round-trip before the action completes.

##### Types — at a glance

All declared in `libs/ledger-live-common/src/device/use-cases/ensureAppReady/types.ts` (except the state, which lives in `libs/live-dmk-shared/src/device-action/EnsureAppReady/state.ts`):

- **Flow input** (caller-provided, legacy-compatible): `AppRequestInput` — `account` / `currency` / `tokenCurrency` / `dependencies` / `requireLatestFirmware` / `allowPartialDependencies`.
- **Use-case input** (DIE-shaped, the boundary the use case actually consumes): `EnsureAppReadyInput` — `appName`, `dependencies`, `requireLatestFirmware`, `allowPartialDependencies`, `requiresDerivation?`, `expectedAccount?`, `deprecation?`. Built from `AppRequestInput` by `buildEnsureAppReadyInput`.
- **Dependencies** (injected, defaultable): `EnsureAppReadyUseCaseDependencies` (declared in `ensureAppReadyUseCase.ts`) — `getMinVersion`, `getDeprecationConfig`, `shouldUpgrade`. Defaults point to the real Live implementations from `libs/ledger-live-common/src/apps/support.ts`; tests and the debug screen override them per-scenario.
- **Side effects** (host-app provided): `ConnectAppInitSideEffects` — `onDeviceIdObserved`, `onLastSeenDeviceInfoObserved`. The use case itself stays pure; the host (LWM) wires these to its store actions in `useDeviceContextInitializerComponentLWMViewModel.ts`.
- **Output**: `Observable<EnsureAppReadyState>` — exhaustive UI states, terminating on a final state (`Success` / `Error` / blocking).

##### Responsibilities — at a glance

| Layer | File | Responsibility |
| --- | --- | --- |
| `buildEnsureAppReadyInput` | `libs/ledger-live-common/src/device/use-cases/ensureAppReady/buildEnsureAppReadyInput.ts` | Translation boundary: `AppRequestInput` → `EnsureAppReadyInput`. Reuses `resolveAppRequestRequirements` (shared with legacy). |
| `ensureAppReadyUseCase` | `libs/ledger-live-common/src/device/use-cases/ensureAppReady/ensureAppReadyUseCase.ts` (+ `helpers/runEnsureAppReadyAttempt.ts`, `helpers/withRetryableRepeat.ts`) | Wires DMK + injected config + side effects + retry into a single `Observable<EnsureAppReadyState>` the UI can render directly. |
| `buildConnectAppDeviceActionInput` | `libs/ledger-live-common/src/device/use-cases/ensureAppReady/helpers/buildConnectAppDAInput.ts` | Builds the `ConnectAppDAInput` (per-model min-version constraints, deprecation config, required-derivation closure) from the injected dependencies. |
| `EnsureAppReadyDeviceAction` | `libs/live-dmk-shared/src/device-action/EnsureAppReady/EnsureAppReadyDeviceAction.ts` | DMK-native orchestration around `ConnectAppDeviceAction`: snapshot → state mapping, outdated-app warning, final-state emission. |
| `EnsureAppReadyStateEmitter` + `stateMapping` | `libs/live-dmk-shared/src/device-action/EnsureAppReady/{EnsureAppReadyStateEmitter,stateMapping}.ts` | Pure mapping from `ConnectAppDAState` snapshots/errors to `EnsureAppReadyState`. |
| `ConnectAppDeviceAction` | `libs/live-dmk-shared/src/device-action/ConnectApp/ConnectAppDeviceAction.ts` | Existing DMK device action: open / install / update app, run required derivation, surface device interactions, emit deprecation rules. **Extended in this PR** with a `RefreshDeviceStatus` step and `deviceMetadata` exposure in `intermediateValue` — see [Changes to `ConnectAppDeviceAction`](#changes-to-connectappdeviceaction). |
| `ConnectAppSideEffectsHandler` | `libs/ledger-live-common/src/device/use-cases/ensureAppReady/helpers/ConnectAppSideEffectsHandler.ts` | Live-specific side effects (identities slice, last-seen device info). |
| `buildFinalState` | `libs/ledger-live-common/src/device/use-cases/ensureAppReady/helpers/buildFinalState.ts` | Wrong-device-for-account check using Live account identity rules. |
| `DeviceContextInitializerComponentLWM` | `apps/ledger-live-mobile/src/mvvm/components/DeviceIntentExecutor/DeviceContextInitializerComponentLWM/` | Subscribes to the use case, renders one screen per state, forwards the extracted context to the next step. |

#### Design choices

- **Single shared use case, UI-ready state stream**. All policy lives in `ensureAppReadyUseCase`; the UI just runs it and renders. This avoids the legacy "one big mega-state for everything" problem and keeps host apps (LWM today, more later) in sync with zero duplication.
- **Reuse `ConnectAppDeviceAction`** as the orchestration core. It already handles connect / open / install / update / required-derivation. We wrap it with `EnsureAppReadyDeviceAction` instead of reimplementing it, which keeps the surface small. The two small additions made to it (`RefreshDeviceStatus` step + `deviceMetadata` in `intermediateValue`) are scoped and additive — see [Changes to `ConnectAppDeviceAction`](#changes-to-connectappdeviceaction).
- **Configuration is injected, not pulled**. `getMinVersion`, `getDeprecationConfig`, `shouldUpgrade` are dependencies with sensible production defaults. Tests and the debug screen override them in-process — no remote config mutation needed to exercise edge cases.
- **Full parity with the legacy `connectApp` flow**:
  - same input shape (`AppRequestInput`) so flow migration is a one-liner translation,
  - same side effects (identities slice + last-seen device info),
  - same domain checks (deprecation, min app version, outdated app, wrong device for account, derivation requirement),
  - first-class states for everything the legacy flow used to expose as errors (`WrongDeviceForAccount`, `DeviceDeprecatedBlocking`, `UnsupportedApplication`, `UnsupportedFirmwareVersion`, `UnsupportedFeature`, `DeviceOutOfStorageSpace`, `DeviceNotOnboarded`).
- **Retry is part of the state contract**. Retryable states carry their own `retry()` callback. The view doesn't decide what is retryable.

#### LWM-side wiring (validation only — not the final UI)

Folder: `apps/ledger-live-mobile/src/mvvm/components/DeviceIntentExecutor/DeviceContextInitializerComponentLWM/`

- `index.tsx` — `DeviceContextInitializerComponent`-conforming entry point that picks up the optional injected `dependencies` from the executor's `InitializerConfig`.
- `useDeviceContextInitializerComponentLWMViewModel.ts` — runs `ensureAppReadyUseCase` in `useEffect`, stores each state in `useState`, wires the LWM-specific side effects (`identitiesSlice.actions.addDeviceId`, `setLastSeenDeviceInfo`) and forwards the extracted context to `onContextInitialized` on `Success`.
- `DeviceContextInitializerComponentLWMView.tsx` — pure presentational view, single `switch` over every `EnsureAppReadyState.type`.

This is intentionally a **debug-quality UI**. Its purpose is functional validation of the new flow, not the final shipping UX. The polished UI will land in a follow-up.

#### Debug screen — Device Intent Executor → Initialization Playground

Folder: `apps/ledger-live-mobile/src/screens/Settings/Debug/Features/DeviceIntentExecutor/`

A new debug screen (`InitializationScreen.tsx`) has been added at **Settings → Debug → Device Intent Executor → Initialization** that wires the real DIE initializer + a tiny "echo" intent (`intents/initializationEchoIntent/`) that just prints the device context the job received once initialization succeeded.

Scenarios are declared in `initializationScenarios.ts`, each providing both an `EnsureAppReadyInput` and an optional `InitializerConfig` (used to override the use case's injected dependencies per-scenario, so we can exercise edge cases without touching any remote config):

- BOLOS dashboard (no derivation),
- Ethereum derivation (`44'/60'/0'/0/0`),
- Bitcoin derivation (native segwit `84'/0'/0'/0/0`),
- Bitcoin forced `shouldUpgrade()` → `true` (LSB-010 segwit upgrade path),
- Ethereum forced `getMinVersion() = 99.99.99` (forces install/update path),
- Ethereum send deprecation **warning** screen (past warning date),
- Ethereum send deprecation **clear-signing** warning screen,
- Ethereum send deprecation **warning + clear-signing** combined,
- Ethereum send deprecation **blocking** error screen,
- Ethereum **wrong-account** (impossible expected address — exercises the wrong-device-for-account guard).

This screen is also used as the QA entry point for this PR.

### ❓ Context

- **JIRA or GitHub link**: [LIVE-29431](https://ledgerhq.atlassian.net/browse/LIVE-29431) [LIVE-29432](https://ledgerhq.atlassian.net/browse/LIVE-29432)
- **ADR link (if any)**: _N/A_

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-29431]: https://ledgerhq.atlassian.net/browse/LIVE-29431?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/live-dmk-shared/src/device-action/EnsureAppReady/EnsureAppReadyStateEmitter.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      state,
+      appName: this.params.appName,
+      deprecation: this.params.deprecation,
+      deprecationDismissedCurrencyNames: this.params.deprecationDismissedCurrencyNames,
+    });
+
+    log("[EnsureAppReady]", "EnsureAppReadyStateEmitter: handlePendingState", {
+      state,
+      mappedState,
+    });
```
[the remark this answers is not in the document]
Comment (OlivierFreyssinet) [reply libs/live-dmk-shared/src/device-action/EnsureAppReady/EnsureAppReadyStateEmitter.ts:78]: useful for now

File: apps/ledger-live-mobile/src/mvvm/components/DeviceIntentExecutor/DeviceContextInitializerComponentLWM/__tests__/useDeviceContextInitializerComponentLWMViewModel.test.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
+import { Observable, Subject } from "rxjs";
+import { act, renderHook } from "@testing-library/react-native";
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/mvvm/components/DeviceIntentExecutor/DeviceContextInitializerComponentLWM/__tests__/useDeviceContextInitializerComponentLWMViewModel.test.tsx]: using `renderHook` from  `test-renderer` will allows you to not mock  `jest.mock("~/context/hooks", () => ({`

**Review by benruseau — APPROVED** [review]

File: apps/ledger-live-mobile/src/mvvm/components/DeviceIntentExecutor/DeviceContextInitializerComponentLWM/__tests__/useDeviceContextInitializerComponentLWMViewModel.test.tsx (hunk 2 above)
Comment (OlivierFreyssinet) [reply apps/ledger-live-mobile/src/mvvm/components/DeviceIntentExecutor/DeviceContextInitializerComponentLWM/__tests__/useDeviceContextInitializerComponentLWMViewModel.test.tsx]: done 👍

**Review by thesan — APPROVED** [review]

**Review by mcayuelas-ledger — APPROVED** [review]
